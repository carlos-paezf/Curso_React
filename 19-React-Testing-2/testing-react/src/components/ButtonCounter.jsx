import React from 'react'

const ButtonCounter = ({ name, value, action, aria}) => {
    return (
        <button aria-label={aria} onClick={() => action(value)}>{name}</button>
    )
}

export default ButtonCounter
