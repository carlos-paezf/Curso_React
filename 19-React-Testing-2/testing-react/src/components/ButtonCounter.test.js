import { render, screen } from '@testing-library/react'
import ButtonCounter from './ButtonCounter'

describe('<ButtonCounter />', () => {

    const action = jest.fn()

    beforeEach(() => {
        render(<ButtonCounter name="Prueba" value={3} action={action} aria="prueba-aumentar" />)
    })

    it('Snapshot para <ButtonCounter />', () => {
        expect(screen.getByLabelText('prueba-aumentar')).toMatchSnapshot()
    })

    it('Crear un botÃ³n correctamente', () => {
        expect(screen.getByLabelText('prueba-aumentar')).toBeInTheDocument()
    })
})
