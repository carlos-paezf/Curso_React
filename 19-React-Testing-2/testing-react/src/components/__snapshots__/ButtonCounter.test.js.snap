// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonCounter /> Snapshot para <ButtonCounter /> 1`] = `
<button
  aria-label="prueba-aumentar"
>
  Prueba
</button>
`;
